runtime/Cpp/runtime/src/Exceptions.cpp

```diff
@@ -22,3 +22,12 @@ IOException::IOException(const std::string &msg) : std::exception(), _message(ms
 const char* IOException::what() const NOEXCEPT {
   return _message.c_str();
 }
+
+IllegalStateException::~IllegalStateException() = default;
+IllegalArgumentException::~IllegalArgumentException() = default;
+NullPointerException::~NullPointerException() = default;
+IndexOutOfBoundsException::~IndexOutOfBoundsException() = default;
+UnsupportedOperationException::~UnsupportedOperationException() = default;
+EmptyStackException::~EmptyStackException() = default;
+CancellationException::~CancellationException() = default;
+ParseCancellationException::~ParseCancellationException() = default;
```

Comment:
The problem with putting the "= default" in the header file is that the vtable is not generated as part of the translation unit. I kept the same order as the pattern used in the header file changes, the difference is that the destructor doesn't have an inline definition.

Would you prefer an empty destructor or "= default" in a .cpp file?